src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,225 @@
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMax, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+            else:
+                logging.info("HoldTime conformance failed.  Test step skipped")
+
+        else:
+            logging.info("HoldTimeLimits not supported. Test step skipped")
+        
+        self.step(6)
+        if attributes.HoldTime.attribute_id in attribute_list:
+            hold_time_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+
+            asserts.assert_less_equal(hold_time_dut, 0xFFFE, "HoldTime attribute is out of range")
+            asserts.assert_greater_equal(hold_time_dut, 0, "HoldTime attribute is out of range")
+        else:
+            logging.info("HoldTime not supported. The rest of legacy attribute test can be skipped")
+
+        self.step(7)
+        if attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list:
+            if ((occupancy_sensor_type_bitmap_dut == 0b00000001) | ((occupancy_sensor_type_bitmap_dut != 0b00000001)&(occupancy_sensor_type_bitmap_dut != 0b00000010)&(occupancy_sensor_type_bitmap_dut != 0b00000100))):
+                pir_otou_delay_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIROccupiedToUnoccupiedDelay)
+
+                asserts.assert_less_equal(pir_otou_delay_dut, 0xFFFE, "PIROccupiedToUnoccupiedDelay is not in valid range")
+                asserts.assert_greater_equal(pir_otou_delay_dut, 0, "PIROccupiedToUnoccupiedDelay is not in valid range")
+            else:
+                logging.info("PIROccupiedToUnoccupiedDelay conformance failed. Test step skipped")
+        else:
+            logging.info("PIROccupiedToUnoccupiedDelay not supported. Test step skipped")
+
+        self.step(8)
+        if attributes.PIRUnoccupiedToOccupiedDelay.attribute_id in attribute_list: 
+            if attributes.PIRUnoccupiedToOccupiedThreshold.attribute_id in attribute_list:
+            
+                pir_utoo_delay_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIRUnoccupiedToOccupiedDelay)
+                asserts.assert_less_equal(pir_utoo_delay_dut, 0xFFFE, "PIRUnoccupiedToOccupiedDelay is not in valid range")
+                asserts.assert_greater_equal(pir_utoo_delay_dut, 0, "PIRUnoccupiedToOccupiedDelay is not in valid range")
+            else:
+                logging.info("PIRUnoccupiedToOccupiedDelay conformance failed. Test step skipped")

```

Comment:
See above - this doesn't actually fail the test.

Reply:
assert implementation will take care of this.